### ISSUE TITLE
Format of CloudFormation template that uses Fn::Transform yields invalid template

### ISSUE THREAD

**Issue by tomirio619** [aws-cloudformation/rain#59]
I have the following CloudFormation snippet called `input.yml`:

```yaml
Resources:
  RestApi:
    Type: AWS::ApiGateway::RestApi
    Properties:
      ApiKeySourceType: HEADER
      Body:
        Fn::Transform:
          Name: AWS::Include
          Parameters:
            Location: s3://<bucket-name>/<object-key>
      Description: API
      EndpointConfiguration:
        Types:
          - PRIVATE
      FailOnWarnings: true
      Name: My API
```

When I run `$ rain fmt input.yml > output.yml`, the contents of `output.yml` are:
```yaml
AWSTemplateFormatVersion: "2010-09-09"

Resources:
  RestApi:
    Type: AWS::ApiGateway::RestApi
    Properties:
      ApiKeySourceType: HEADER
      Body: !Transform
        Name: AWS::Include
        Parameters:
          Location: s3://<bucket-name>/<object-key>
      Description: My API
      EndpointConfiguration:
        Types:
          - PRIVATE
      FailOnWarnings: true
      Name: API
```
If I deploy this formatted template in CloudFormation, deployment fails with the following message:
```
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template format error: YAML not well-formed. 
```

Unfortunately, I cannot make use of `S3BodyLocation` (instead of `Body`), as our OpenAPI specification contains CloudFormation functions which are not substituted when used with `S3BodyLocation` (see [this issue](https://github.com/aws/serverless-application-model/issues/79#issuecomment-435044422)).
If used, the following message is shown for the `CREATE_FAILED` status:
```
Unable to parse API definition because of a malformed integration at path /an/example/path. (Service: AmazonApiGateway; Status Code: 400; Error Code: BadRequestException; Request ID: REDACTED; Proxy: null)
```

The [`AWS::Include` transform](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/create-reusable-transform-function-snippets-and-add-to-your-template-with-aws-include-transform.html) documentation could be related, as it mentions the following:
```
We don't currently support using shorthand notations for YAML snippets.
```

**Comment by stilvoid** [comment]
Thanks for this! After some digging, it seems the Transform function just doesn't support the YAML tag-style notation. I will prevent Fn::Transform from being modified by `rain fmt` in the next release.

**Comment by stilvoid** [comment]
Fixed in dd5001e09da5550bfcb9d3e69892d1561c864829

**Comment by tomirio619** [comment]
Great, thanks!
